src/Arcus.WebApi.Logging/ExceptionHandlingMiddleware.cs

```diff
@@ -62,31 +62,48 @@ public async Task Invoke(HttpContext context, ILoggerFactory loggerFactory)
             {
                 await _next(context);
             }
+            catch (Exception exception)
+            {
+                LogException(loggerFactory, exception);
+
+                HttpStatusCode statusCode = DetermineResponseStatusCode(exception);
+                context.Response.StatusCode = (int) statusCode;
+            }
+        }
+
+        /// <summary>
+        /// Logs the caught <paramref name="exception"/> before the response it sent back.
+        /// </summary>
+        /// <param name="loggerFactory">The factory instance to create <see cref="ILogger"/> instances, specifically for the <paramref name="exception"/>.</param>
+        /// <param name="exception">The caught exception during the application pipeline.</param>
+        protected virtual void LogException(ILoggerFactory loggerFactory, Exception exception)
```

Comment:
Wouldn't it be easier for consumers to pass in an `ILogger` already, instead of the `ILoggerFactory`?

Reply:
Ah, yes, it would be. 😄 Was wonder how much control users would want.

Reply:
Done in https://github.com/arcus-azure/arcus.webapi/pull/342/commits/5e59eb6082bc01739b9fc13a2195e8474614d505

Reply:
I think in this case, users prefer simplicity :)